app/assets/javascripts/host_edit.js

```diff
@@ -77,6 +75,22 @@ function computeResourceSelected(item){
   }
 }
 
+function handle_nic_updates() {
+  var modal_window = $('#interfaceModal');
+
+  var handler = function(event = null) {
+    update_nics(function() {
+      interface_subnet_selected(primary_nic_form().find('select.interface_subnet'));
+    });
+  };
+
+  if (modal_window.is(":visible")) {
+    modal_window.modal('hide').on('hidden.bs.modal', handler);
```

Comment:
> The modal window div gets destroyed

where does this happen? I think it's only hidden and it's body gets cleared. is `.modal('hide')` async? maybe just this would work
```
modal_window.modal('hide')
handler
```